clustering/ejb/WFLY-14359_Create_distributable-ejb_subsystem.adoc

```diff
@@ -0,0 +1,188 @@
+
+=== Nice-to-Have Requirements
+
+The ability to allow overriding the default bean manager for a whole deployment is not a hard requirement, but a nice-to-have.
+
+=== Non-Requirements
+
+== Backwards Compatibility
+
+These changes affect backward compatability (c.f. legacy configuration of cache factories, passivation stores,
+client mappings registries, etc)
+
+=== Default Configuration
+
+Here is an example of the way the new subsystem will look, for the case of specifying a bean management provider:
+----
+<subsystem xmlns="urn:jboss:domain:distributable-ejb:1.0" default-bean-management="infinispan">
+    <infinispan-bean-management name="infinispan" cache-container="ejb" cache="dist" max-size="10"/>
+</subsystem>
+----
+and here is its corresponding integration into the ejb3 subsystem:
+----
+<subsystem xmlns="urn:jboss:domain:ejb3:10.0">
+     ...
+     <caches>
+        <simple-cache name="simple-cache"/>
+        <distributable-cache name="distributable-cache" bean-management="infinispan"/>
+    </caches>
+    ...
+ </subsystem>
+----
+Note that the distributable-cache element takes an optional bean-management attribute, which is a reference to
+the bean management provider in the new subsystem. A similar arrangment will be used for client mappings registries.
+When unspecified, the default-bean-management profile is used.
+
+As before, for each EJB SFSB, the @Cache annotation (resp. deployment descriptor configuration) may be used to specify
+the name of the desired cache factory defined in the ejb3 subsystem. If no @Cache annotation (resp. deployment descriptor
+configuration) is present, a suitable default cache will be used for the bean.
+
+=== Importing Existing Configuration
+
+=== Deployments
+
+A new deployment configuration namespace, specified via jboss-all.xml or a separate distributable-ejb.xml, will be introduced
+to permit specifying default providers for bean management (resp. client-mappings management) on a per-deployment basis.
+This means that named, configured providers for bean management (resp. client-mappings management) may be specified in
+several places:
+
+* in the distributable-ejb subsystem, providing server-wide default provider values
+* in the distributable-ejb.xml file included with a deployment, providing deployment-scoped default provider values
+
+=== Interoperability
+
+== Implementation Plan
+
+As mentioned in the overview, the distributable-ejb subsystem will eventually support the provision of providers for four
+key areas of ejb3 subsystem functionality. However, this issue will be restricted to the following scope:
+
+* defining the distributable-ejb subsystem itself
+* defining a cache factory provider for stateful session beans and integrating that provider into the ejb3 subsystem
+* defining a client mappings registry provider for the EJB client and integrating that provider into the ejb3 subsystem
+
```

Comment:
> it be useful to mention some more low level details

The intention of the "Implementation plan" section is for RFEs with sufficiently complex implementations for which other engineers may want to provide feedback, e.g. WFLY-7628.  This section is not applicable to this RFE.

Reply:
Thanks @pferraro - You're right and I've read the hints in the template, but since it was filled in I thought that some _actual_ implementation details would have been a value add.